### DISCUSSION TITLE
JDBC client / server version compatibility matrix

### DISCUSSION THREAD

**Discussion by tooptoop4** [trinodb/trino#14159] [Q&A] [answered]
I have protocol.v1.alternate-header-name=Presto

would be great to have a table like this (but **I'm not sure the right version values**...):

JDBC server version range | JDBC client version range
--- | ---
<=301 | 220-303 
302-316 | 303-320
317-353 | 310-360
 +354 | >=352

I vaguely recall timestamp types were changing to have a zone or not and also timestamp precision changing but not sure if the clients had backwards compatibility

**Comment by findepi** [comment, marked as answer]
Currently, every Trino server supports all JDBC drivers >= 352
https://github.com/trinodb/trino/blob/029d9c0dd5326dade3654bf2a67e9a8609a61d94/testing/trino-test-jdbc-compatibility-old-driver/bin/run_tests.sh#L13

Also, the current JDBC driver supports Trino servers 15 releases back
https://github.com/trinodb/trino/blob/c6373462dedf2fb356c3ed70ccdc278a09073229/testing/trino-test-jdbc-compatibility-old-server/src/test/java/io/trino/TestJdbcResultSetCompatibilityOldServer.java#L72-L79
(eg for current release 395, the oldest Trino server supported is 380).

You can derive the information you need from the information above.
However, since we don't "maintain" old versions, adding actual compatibility ranges to docs wouldn't be appropriate IMO.

This project runs releases off of master, with no maintenance branches and has no capacity to maintain/document information about old releases.
